File: c/LeetCode_48_RotateImage.c
#include <stdio.h>
#include <stdlib.h>


void rotate(int** matrix, int matrixSize, int* matrixColSize) {
    for (int i = 0; i < matrixSize; ++i) {
        for (int j = i + 1; j < matrixSize; ++j) {
            int tmp = matrix[i][j];
            matrix[i][j] = matrix[j][i];
            matrix[j][i] = tmp;
        }
    }

    int hlf = matrixSize / 2;
    for (int i = 0; i < matrixSize; ++i) {
        int inx = matrixSize - 1;
        for (int j = 0; j < hlf; ++j) {
            int tmp = matrix[i][j];
            matrix[i][j] = matrix[i][inx];
            matrix[i][inx] = tmp;
            --inx;
        }
    }
}

File: python/LeetCode_48_RotateImage.py
def rotate(matrix) -> None:
    lngth = len(matrix)
    for i in range(lngth):
        for j in range(i + 1, lngth):
            matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]


    hlf = (lngth // 2)
    for i in range(lngth):
        inx = lngth - 1
        for j in range(hlf):
            matrix[i][j], matrix[i][inx] = matrix[i][inx], matrix[i][j]
            inx -= 1


     



rotate([[1,2,3],[4,5,6],[7,8,9]])